packages/std/src/mock.rs

```diff
@@ -110,29 +123,34 @@ pub fn mock_env<T: Api, U: Into<HumanAddr>>(
 
 /// MockQuerier holds an immutable table of bank balances
 /// TODO: also allow querying contracts
+#[derive(Clone)]
 pub struct MockQuerier {
     balances: HashMap<HumanAddr, Vec<Coin>>,
 }
 
 impl MockQuerier {
-    pub fn new(balances: HashMap<HumanAddr, Vec<Coin>>) -> Self {
-        MockQuerier { balances }
+    pub fn new(balances: Vec<(HumanAddr, Vec<Coin>)>) -> Self {
```

Comment:
How do you feel about passing this as a read only reference (is slice of references) and then cloning it into the internal data store? I think it would simplify a lot of callers that don't need to clone at call time.